server/jetty/src/main/java/io/deephaven/server/jetty/ControlledCacheResource.java

```diff
@@ -0,0 +1,122 @@
+package io.deephaven.server.jetty;
+
+import org.eclipse.jetty.util.resource.Resource;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.nio.channels.ReadableByteChannel;
+
+/**
+ * Simple wrapper around the Jetty Resource type, to grant us control over caching features. The current implementation
+ * only removes the last-modified value, but a future version could provide a "real" weak/strong etag.
+ */
+public class ControlledCacheResource extends Resource {
+    private final Resource wrapped;
+
+    public ControlledCacheResource(Resource wrapped) {
+        this.wrapped = wrapped;
+    }
+
+    @Override
+    public boolean isContainedIn(Resource r) throws MalformedURLException {
+        return wrapped.isContainedIn(r);
+    }
+
+    @Override
+    public void close() {
+        wrapped.close();
+    }
+
+    @Override
+    public boolean exists() {
+        return wrapped.exists();
+    }
+
+    @Override
+    public boolean isDirectory() {
+        return wrapped.isDirectory();
+    }
+
+    @Override
+    public long lastModified() {
+        // Always return -1, so that we don't get the build system timestamp. In theory we could return the app startup
+        // time as well, so that clients that connect don't need to revalidate quite as often, but this could have other
+        // side effects such as in load balancing with a short-lived old build against a seconds-older new build.
+        return -1;
+    }
+
+    @Override
+    public long length() {
+        return wrapped.length();
+    }
+
+    @Override
+    public URI getURI() {
+        return wrapped.getURI();
+    }
+
+    @Override
+    public File getFile() throws IOException {
+        return wrapped.getFile();
+    }
+
+    @Override
+    public String getName() {
+        return wrapped.getName();
+    }
+
+    @Override
+    public InputStream getInputStream() throws IOException {
+        return wrapped.getInputStream();
+    }
+
+    @Override
+    public ReadableByteChannel getReadableByteChannel() throws IOException {
+        return wrapped.getReadableByteChannel();
+    }
+
+    @Override
+    public boolean delete() throws SecurityException {
+        return wrapped.delete();
+    }
+
+    @Override
+    public boolean renameTo(Resource dest) throws SecurityException {
+        return wrapped.renameTo(dest);
+    }
+
+    @Override
+    public String[] list() {
+        return wrapped.list();
+    }
+
+    @Override
+    public Resource addPath(String path) throws IOException, MalformedURLException {
+        // Re-wrap any instance that might be returned
+        return new ControlledCacheResource(wrapped.addPath(path));
```

Comment:
Might make sense to check if `wrapped instanceof ControlledCacheResource` so we can reduce re-wrapping a bunch of times? Not sure if issue in practice, but something like:

```java
cacheResource.addPath("1").addPath("2").addPath("3")...
```

Reply:
Probably makes sense to test alongside the JS plugin PR which creates zip resources.

Reply:
I can add that check, but each addPath will only wrap the next layer, right?
 * `cacheResource` is singly wrapped
 * `resource.addPath("1")` returns a singly wrapped instance, `resource1`
 * `resource1.addPath("2")` returns a singly wrapped instance, etc
 
Unless the initially created ControlledCacheResource is multiply wrapped to begin with, _or_ the wrapped resource itself somehow contains already-wrapped resources (for example a ResourceCollection full of already wrapped resources, and then wrap the ResourceCollection again), I can't see a way for this to happen? What am I missing?

Mitigation of the theoretical issue will be replacing the public ctor with a static wrap() method that does the check, and delegate to that when wrapping a sub-resource.